WcaOnRails/app/models/person.rb

```diff
@@ -167,11 +167,64 @@ def best_solve(event, type)
   end
 
   def world_championship_podiums
-    results.includes(:competition, :event, :format)
-           .joins(competition: [:championships])
-           .podium
+    results.podium
+           .joins(:event, competition: [:championships])
            .where("championships.championship_type = 'world'")
            .order("year DESC, Events.rank")
+           .includes(:competition, :format)
+  end
+
+  def championship_podiums_with_condition
+    # Get all championship competitions of the given type where the person made it to the finals.
+    # For each of these competitions, get final results only for people eligible for the championship
+    # and reassign their positions. If a result belongs to the person, add it to the array.
+    [].tap do |championship_podium_results|
+      yield(results)
+        .final
+        .succeeded
+        .order("year DESC")
+        .includes(:competition)
+        .map(&:competition)
+        .uniq
+        .each do |competition|
+          yield(competition.results)
+            .final
+            .succeeded
+            .joins(:event)
+            .order("Events.rank, pos")
+            .includes(:format, :competition)
+            .group_by(&:eventId)
+            .each_value do |final_results|
+              previous_old_pos = nil
+              previous_new_pos = nil
```

Comment:
I'm confused, will `previous_new_pos` or `previous_old_pos` ever get set to anything other than `nil`?

Could you add a test that checks for ties, to test the case that @viroulep pointed out?

Reply:
Fixed and added spec, sorry...

Reply:
Looks great now. This reminds me of your very first PR to the website =) #437 =) =)